planner/types.ts

```diff
@@ -0,0 +1,161 @@
+export class Activity {
+    name: string;
+    duration: number;
+    NoP: number;
+    role: Role;
+    input: IOSet[];
+    output: IOSet[];
+
+    public constructor(name: string, duration: number, NoP: number, role: Role, input: IOSet[], output: IOSet[]) {
```

Comment:
You could already add default values for the duration (1) and the NoP (1) and maybe null for the role.